src/consts.rs

```diff
@@ -311,13 +332,32 @@ impl<'c, 'cc> ConstEvalContext<'c, 'cc> {
                     UnNeg => constant_negate(o),
                     UnUniq | UnDeref => Some(o),
                 }),
-            ExprBinary(op, ref left, ref right) =>
-                self.binop(op, left, right),
+            ExprBinary(op, ref left, ref right) => self.binop(op, left, right),
+            ExprCast(ref value, ref ty) => self.cast(value, ty),
+            ExprIndex(ref x, ref index) =>
+                if let ExprVec(ref v) = x.node {
+                    self.expr(index).and_then(
+                        |i| self.index(v, i.as_u64() as usize) )
+                } else { None },
+            ExprTupField(ref x, ref i) =>
+                if let ExprTup(ref t) = x.node {
+                    self.index(t, i.node)
+                } else { None },
             //TODO: add other expressions
+            //ExprStruct(Path, Vec<Field>, Option<P<Expr>>),
+            //ExprField(P<Expr>, SpannedIdent),
+            //ExprRange(Option<P<Expr>>, Option<P<Expr>>),
```

Comment:
But a combination of struct-field could.

Note to self: I may want to do some lookups/paren or block extraction/etc. in between.